aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java

```diff
@@ -136,15 +136,20 @@ private <T extends Model> Completable processOutboxItem(PendingMutation<T> mutat
             .andThen(publishToNetwork(mutationOutboxItem)
                 .map(modelWithMetadata -> ensureModelHasSchema(mutationOutboxItem, modelWithMetadata))
                 .flatMapCompletable(modelWithMetadata ->
-                    // Once the server knows about it, it's safe to remove from the outbox.
-                    // This is done before merging, because the merger will refuse to merge
-                    // if there are outstanding mutations in the outbox.
-                    mutationOutbox.remove(mutationOutboxItem.getMutationId())
-                        .andThen(merger.merge(modelWithMetadata))
-                        .doOnComplete(() -> {
-                            String modelName = mutationOutboxItem.getModelSchema().getName();
-                            announceMutationProcessed(modelName, modelWithMetadata);
-                        })
+                            // Once the server knows about it, it's safe to remove from the outbox.
```

Comment:
nit: looks like the indentation is off here

Reply:
Resolved